internal/controller/serverset/serverset.go

```diff
@@ -467,38 +471,44 @@ func (e *external) Delete(ctx context.Context, mg resource.Managed) error {
 	return nil
 }
 
-// AreServersUpToDate checks if replicas and template params are equal to server obj params
-func AreServersUpToDate(templateParams v1alpha1.ServerSetTemplateSpec, servers []v1alpha1.Server) bool {
+// AreServersReady checks if replicas and template params are equal to server obj params
+func AreServersReady(templateParams v1alpha1.ServerSetTemplateSpec, servers []v1alpha1.Server) (areServersUpToDate, areServersAvailable bool) {
 	for _, serverObj := range servers {
 		if serverObj.Spec.ForProvider.Cores != templateParams.Cores {
-			return false
+			return false, false
 		}
 		if serverObj.Spec.ForProvider.RAM != templateParams.RAM {
-			return false
+			return false, false
 		}
 		if serverObj.Spec.ForProvider.CPUFamily != templateParams.CPUFamily {
-			return false
+			return false, false
+		}
+		if serverObj.Status.AtProvider.State != ionoscloud.Available {
+			return true, false
 		}
 	}
 
-	return true
+	return true, true
 }
 
-// AreVolumesUpToDate checks if template params are equal to volume obj params
-func AreVolumesUpToDate(templateParams v1alpha1.BootVolumeTemplate, volumes []v1alpha1.Volume) bool {
+// AreVolumesReady checks if template params are equal to volume obj params
+func AreVolumesReady(templateParams v1alpha1.BootVolumeTemplate, volumes []v1alpha1.Volume) (bool, bool) {
```

Comment:
Maybe we could say here `AreBootVolumesReady` so that we know it is about the boot volumes only?